frame/composable-support/src/validation.rs

```diff
@@ -33,64 +95,77 @@ pub struct Valid;
 #[derive(Debug, Eq, PartialEq, Default)]
 pub struct Invalid;
 
-impl<T> Validate<Invalid> for T {
+impl<T> Validate<T, Invalid> for Invalid {
 	#[inline(always)]
-	fn validate(self) -> Result<Self, &'static str> {
+	fn validate(_extinsic_input: T) -> Result<T, &'static str> {
 		Err("not valid")
 	}
 }
 
-impl<T> Validate<Valid> for T {
+impl<T> Validate<T, Valid> for Valid {
 	#[inline(always)]
-	fn validate(self) -> Result<Self, &'static str> {
-		Ok(self)
+	fn validate(extrinsic_input: T) -> Result<T, &'static str> {
+		Ok(extrinsic_input)
 	}
 }
 
-impl<T: Validate<U> + Validate<V>, U, V> Validate<(U, V)> for T {
+impl<T, U, V> Validate<T, (U, V)> for (U, V)
+where
+	U: Validate<T, U>,
+	V: Validate<T, V>,
+{
 	#[inline(always)]
-	fn validate(self) -> Result<Self, &'static str> {
-		let value = Validate::<U>::validate(self)?;
-		let value = Validate::<V>::validate(value)?;
+	fn validate(extrinsic_input: T) -> Result<T, &'static str> {
+		let value = <U as Validate<T, U>>::validate(extrinsic_input)?;
+		let value = <V as Validate<T, V>>::validate(value)?;
 		Ok(value)
 	}
 }
 
 // as per substrate pattern and existing macroses for similar purposes, they tend to make things
 // flat like `#[impl_trait_for_tuples::impl_for_tuples(30)]`
 // so if we will need more than 3, can consider it
-impl<T: Validate<U> + Validate<V> + Validate<W>, U, V, W> Validate<(U, V, W)> for T {
+impl<T, U, V, W> Validate<T, (U, V, W)> for (U, V, W)
+where
+	U: Validate<T, U>,
+	V: Validate<T, V>,
+	W: Validate<T, W>,
+{
 	#[inline(always)]
-	fn validate(self) -> Result<Self, &'static str> {
-		let value = Validate::<U>::validate(self)?;
-		let value = Validate::<V>::validate(value)?;
-		let value = Validate::<W>::validate(value)?;
+	fn validate(extrinsic_input: T) -> Result<T, &'static str> {
+		let value = <U as Validate<T, U>>::validate(extrinsic_input)?;
+		let value = <V as Validate<T, V>>::validate(value)?;
+		let value = <W as Validate<T, W>>::validate(value)?;
 		Ok(value)
 	}
 }
 
-impl<T: Validate<U> + Validate<V> + Validate<W> + Validate<Z>, U, V, W, Z> Validate<(U, V, W, Z)>
-	for T
+impl<T, U, V, W, Z> Validate<T, (U, V, W, Z)> for (U, V, W, Z)
+where
+	U: Validate<T, U>,
+	V: Validate<T, V>,
+	W: Validate<T, W>,
+	Z: Validate<T, Z>,
 {
 	#[inline(always)]
-	fn validate(self) -> Result<Self, &'static str> {
-		let value = Validate::<U>::validate(self)?;
-		let value = Validate::<V>::validate(value)?;
-		let value = Validate::<W>::validate(value)?;
-		let value = Validate::<Z>::validate(value)?;
+	fn validate(extrinsic_input: T) -> Result<T, &'static str> {
+		let value = <U as Validate<T, U>>::validate(extrinsic_input)?;
```

Comment:
`U::validate(extrinsic_input)` may work

Reply:
Good call, this will clean up the calls quite a bit. Fixed this here 0c38ae60ceefc9a143e3b85cffc314c94ec7c58f